pkg/storage/s3/s3.go

```diff
@@ -36,6 +36,9 @@ func New(s3Conf *config.S3Config, timeout time.Duration, options ...func(*aws.Co
 	const op errors.Op = "s3.New"
 
 	awsConfig := defaults.Config()
+	// remove anonymous credentials from the default config so that
+	// session.NewSession can auto-resolve credentials from role, profile, env etc.
+	awsConfig.Credentials = nil
```

Comment:
Thank you for the fix.

I have a naive question here:
https://github.com/aws/aws-sdk-go/blob/9e52c7bbf58d1f0910d05b24f4ce5728b2fd7c52/aws/session/session.go#L373

also sets defaults.Config so how does this help while merging credentials?

Reply:
Yea you're right `newSession` also sets it but if you'd continue tracing you'd notice that the same function calls `mergeConfigSrcs` here: https://github.com/aws/aws-sdk-go/blob/9e52c7bbf58d1f0910d05b24f4ce5728b2fd7c52/aws/session/session.go#L414 which is where [credentials are being resolved](https://github.com/aws/aws-sdk-go/blob/v1.32.7/aws/session/session.go#L575):

```
# cfg is the one you pointed out and userCfg is the one we're passing to the sdk
if cfg.Credentials == credentials.AnonymousCredentials && userCfg.Credentials == nil {
		creds, err := resolveCredentials(cfg, envCfg, sharedCfg, handlers, sessOpts)
                 ...
}
```

So as you can see from the above it only tries to resolve credentials if the user hadn't passed any credentials.

Reply:
Wow! That is really subtle and a very good find. Lgtm